File: Python/StackLinkedList.py
class StackLinkedList:


    def __init__(self):
        self.top = None;


    def isEmpty(self):
        return self.top == None

    def push(self, data):
        temp = self.top
        self.top = StackNode(data)
        self.top.next = temp
        print(str(data) + " pushed to the top of the stack")


    def pop(self):
        if(self.isEmpty()):
            print("Stack Underflow");
            return float("-inf")
        
        data = self.top.data
        self.top = self.top.next
        return data
    

    def peek(self):
        if(self.isEmpty()):
            print("Stack Underflow");
            return float("-inf")
        return self.top.data
               




class StackNode:

    def __init__(self, data):
        self.next = None;
        self.data = data;

File: Java/StackLinkedList.java
public class StackLinkedList {
    StackNode top;

    StackLinkedList(){
        top = null;
    }

    boolean isEmpty(){
        return (top == null);

    }


    void push(int data){
        StackNode temp = this.top;
        this.top = new StackNode(data);
        this.top.next = temp;
        System.out.println(data + " pushed to stack");
        
    }

    int pop(){
        


        int data = this.top.data;
        this.top = this.top.next;
        return data;

    }

    int peek(){

        if (this.top == null) {
            System.out.println("Stack Underflow");
            return Integer.MIN_VALUE;
        }
        return this.top.data;

    }



}

class StackNode{
    int data;
    StackNode next;

    StackNode(int val){
        this.data =val;
        this.next = null;
    }


}